### PR TITLE
Make PKGProj's restore the baseline project.json

### PR DESCRIPTION
**Pull request by ericstj**
PKGProj build depends on harvesting dlls from previously shipped
packages.  Rather than force every package in the repo to declare the
version of the last-shipped package in project.json we have a central
copy listing everything.  The problem here is that if this project isn't
restored we'll just silently drop assets from the package, which can
cause tests to fail or package build to fail validation.  This also
causes a problem for folks that want to just sync the repo and build
a single project without doing a full restore.

To ensure that we'll always have the baseline restored I set this as the
ProjectJson for the pkgproj and ensure that we run restore during
package build.

Fixes https://github.com/dotnet/corefx/issues/12058

If folks would prefer I could make this a little cleaner by putting some stuff in buildtools.

/cc @weshaggard @hughbe @jamesqo

### CODE REVIEW

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test

**Comment by hughbe** [comment]
Thanks for fixing this

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test

**Comment by ericstj** [comment]
@weshaggard can you please have a look.

File: pkg/baseline/baseline.props [hunk 1]
```diff
@@ -5,5 +5,13 @@
...
     <PackageIndexFile>$(MSBuildThisFileDirectory)..\Microsoft.Private.PackageBaseline\packageIndex.json</PackageIndexFile>
+    
+    <!-- set the project.json file for pkgproj's to point to the project.json in this directory
+         to ensure that PKGProjs will restore harvested packages. -->
+    <ProjectJson>$(MSBuildThisFileDirectory)\project.json</ProjectJson>
+    <ProjectLockJson>$(MSBuildThisFileDirectory)\project.lock.json</ProjectLockJson>
   </PropertyGroup>
+
+  <!-- make RestorePackages run before HarvestStablePackage to ensure we'll restore the baseline before harvesting  -->
+  <Target Name="EnsureRestore" BeforeTargets="HarvestStablePackage" DependsOnTargets="RestorePackages" />
```
Comment (weshaggard) [inline pkg/baseline/baseline.props]: I'm a little worried about the performance of adding this extra restore to every pkgproj but if you do the other item and make RestorePackages incremental then it won't be a big issue.

**Pull request merged**